crates/aiken-lang/src/tipo.rs

```diff
@@ -241,12 +263,21 @@ impl Type {
     }
 
     pub fn is_primitive(&self) -> bool {
-        self.is_bool()
-            || self.is_bytearray()
-            || self.is_int()
-            || self.is_string()
-            || self.is_void()
-            || self.is_data()
+        let uplc_type = self.get_uplc_type();
+        match uplc_type {
+            UplcType::Bool
+            | UplcType::Integer
+            | UplcType::String
+            | UplcType::ByteString
+            | UplcType::Unit
+            | UplcType::Bls12_381G1Element
+            | UplcType::Bls12_381G2Element
+            | UplcType::Bls12_381MlResult => true,
+
+            UplcType::Data if self.is_data() => true,
+            UplcType::Data => false,
```

Comment:
This is confusing. This is orthogonal to this PR/changes but it seems to me that `get_uplc_type` isn't well-defined and shouldn't return `Data` when something isn't _Data_.

Reply:
Constr and Data share the same type of primitive in Aiken

Reply:
I don't think this is the problem here. The `get_uplc_type` function however returns `Data` for things that aren't `Data` (e.g. functions). Which is why this check is necessary. But rather than having this awkward check, we should fix the output domain of `get_uplc_type`.